drivers/include/net/netdev/ble.h

```diff
@@ -0,0 +1,199 @@
+ *
+ * Such a transmission sequence is started by calling either the radio's send
+ * or receive function while the radio is in idle/standby mode.
+ *
+ * Once a transmission sequence is in progress, the next packet to be send, or
+ * the next reception buffer to be used is specified also using the send/recv
+ * functions. They should be called in the `event_callback` right after the
+ * last transmission (RX or TX) was finished.
+ *
+ * The transmission sequence is aborted by calling `netdev_ble_stop(dev)`
+ * (`netdev->set(dev, NETOPT_BLE_CTX, NULL, 0)`). This will put the radio back
+ * into idle/standby mode.
+ *
+ * ## Radio Context
+ *
+ * As BLE uses time sliced channel hopping, the used channel needs to be set
+ * regularly. Additionally, also the used access address and the CRC initializer
+ * need to be set regularly, as they differ for different BLE connections. To
+ * make setting these values more efficient, this interface combines these three
+ * values in to a so called `radio context` and adds a `netopt` option to set
+ * all three values at once using `netdev_ble_set_ctx(dev, ctx)`
+ * (`netdev->set(dev, NETOPT_BLE_CTX, ctx, sizeof(netdev_ble_ctx_t))`).
+ *
+ *
+ * # Implementation Status and Limitations
+ * - This interface works for memory mapped radios only (no support for
+ *   bus-connected devices). This is mainly for timing reasons.
+ * - No support for LE Data Length Extension (bigger packet size), yet
+ *
+ * @file
+ * @brief       BLE specific adaption for the Netdev API
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NET_NETDEV_BLE_H
+#define NET_NETDEV_BLE_H
+
+#include "net/netdev.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Maximum payload length of a standard BLE packet
+ */
+#define NETDEV_BLE_PDU_MAXLEN       (37U)
+
+/**
+ * @brief   Mask for the actual (3 byte) CRC data in the context's CRC field
+ */
+#define NETDEV_BLE_CRC_MASK         (0x00ffffff)
+
+/**
+ * @brief   Flag for marking a correct CRC on packet reception
+ */
+#define NETDEV_BLE_CRC_OK           (0x80000000)
+
+/**
+ * @brief   BLE packet structure (as defined by the BLE standard)
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t flags;                          /**< header flags */
+    uint8_t len;                            /**< actual length of PDU */
+    uint8_t pdu[NETDEV_BLE_PDU_MAXLEN];     /**< protocol data unit (PDU) */
+} netdev_ble_pkt_t;
+
+/**
+ * @brief   Radio context
+ */
+typedef struct {
+    union {
+        uint8_t raw[4];                     /**< byte-wise access */
+        uint32_t u32;                       /**< compact access */
+    } aa;                                   /**< access address */
+    uint32_t crc;                           /**< CRC: 3 LSB for CRC, most
+                                             *   significant bit for RX state*/
+    uint8_t chan;                           /**< channel to use/used */
+} netdev_ble_ctx_t;
```

Comment:
I thought this was explained sufficiently above (see section 'Radio Context').

Reply:
Good documentation is atomic and does not require you to read the whole document to understand one element. At least provide a link to that section, please.

Reply:
This will probably change in the near future anyway again (with Gorm growing up), so I tend to leave it as is and enhance the doc once everything is a little more settled here.

Reply:
OK